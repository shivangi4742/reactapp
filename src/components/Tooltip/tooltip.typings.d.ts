import { ReactNode } from "react";

export type ToolProps = {
    children: ReactNode,
    type?: string,
    size?: string
}