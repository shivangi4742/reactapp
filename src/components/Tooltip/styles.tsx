import Styled, { css } from 'styled-components';
import DS from '../../utils/design-system';
import Tooltip from '@material-ui/core/Tooltip';

const Tooltipstyle = Styled(Tooltip)`
    color: white;
   ${props => props.size == 'sm' && css`
        width: ${DS.TOOLTIPS.SMALL.width}px!important;
        height: ${DS.TOOLTIPS.SMALL.height}px!important;
        font-size: ${DS.TOOLTIPS.SMALL.fontSize}px!important;
   `}
   ${props => props.size == 'lg' && css`
        width: ${DS.TOOLTIPS.LARGE.width}px!important;
        height: ${DS.TOOLTIPS.LARGE.height}px!important;
        font-size: ${DS.TOOLTIPS.LARGE.fontSize}px!important;
   `}
`;

export default Tooltipstyle;