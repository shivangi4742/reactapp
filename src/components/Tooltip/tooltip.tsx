import React from 'react';
import Fab from '@material-ui/core/Fab';
import { ToolProps } from './tooltip.typings';
import Tooltipstyle from './styles';

const Tooltips = ({ children, ...props }: ToolProps) => {

    return (
        <Tooltipstyle title="Add" {...props}  >
            <Fab color="primary" >
                {children}
            </Fab>
        </Tooltipstyle>
    ); 4

}


export default Tooltips;