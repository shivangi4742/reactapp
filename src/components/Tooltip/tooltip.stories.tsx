import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { Grid } from '@material-ui/core';
import Tooltip from './tooltip';
import ComponentAPI from '../component-api';
import ComponentStories from '../component-stories';

const stories = storiesOf('Components', module);

const codeString = `
  <Tooltip size="sm" >
        +
    </Tooltip>
`;

const codeString1 = `
  <Tooltip size="lg" >
        +
    </Tooltip>
`;
const rows = [
    {
        id: 0,
        property: 'size',
        description: 'can be set to small as (sm) large as (lg) or omitted',
        type: 'string',
        default: "-"
    },
];
stories.add(
    'Tooltips',
    (() => <div>
        <h2>Tooltips</h2>
        <p>Tooltips display informative text when users hover over, focus on, or tap an element.</p>
        <h3>Examples <a href="#examples">#</a></h3>
        <Grid container spacing={40}>
            <ComponentStories syntax={codeString} >
                <Tooltip size="sm" >
                    +
                </Tooltip>
            </ComponentStories>
            <ComponentStories syntax={codeString1} >
                <Tooltip size="lg" >
                    +
                </Tooltip>
            </ComponentStories>
        </Grid>
        <ComponentAPI rows={rows} />
    </div>
    )
);