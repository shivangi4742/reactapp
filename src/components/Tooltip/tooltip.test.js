import React from 'react';
import { shallow } from 'enzyme';
import Tooltip from './tooltip';
import toJSON from 'enzyme-to-json';
describe('ToolTip', () => {

 it('should have 1 ToolTip element ', () => {
        const wrapper = shallow(<Tooltip size='sm'>+</Tooltip>);
          console.log(wrapper.debug());
    });
     it('should have 1 ToolTip element ', () => {
        const wrapper = shallow(<Tooltip size='lg'>+</Tooltip>);
          console.log(wrapper.debug());
    });
    it('renders without crashing', () => {
  const tree = shallow(<Tooltip />);
    expect(toJSON(tree)).toMatchSnapshot();
});
});