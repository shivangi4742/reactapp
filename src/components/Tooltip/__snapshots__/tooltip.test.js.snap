// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<Styled(WithStyles(Tooltip))
  title="Add"
>
  <WithStyles(Fab)
    color="primary"
  />
</Styled(WithStyles(Tooltip))>
`;

exports[`ToolTip renders without crashing 1`] = `
<Styled(WithStyles(Tooltip))
  title="Add"
>
  <WithStyles(Fab)
    color="primary"
  />
</Styled(WithStyles(Tooltip))>
`;

exports[`renders without crashing 1`] = `
<Styled(WithStyles(Tooltip))
  title="Add"
>
  <WithStyles(Fab)
    color="primary"
  />
</Styled(WithStyles(Tooltip))>
`;
